command/job_plan.go

```diff
@@ -173,11 +177,75 @@ func (c *JobPlanCommand) Run(args []string) int {
 			c.Colorize().Color(fmt.Sprintf("[bold][yellow]Job Warnings:\n%s[reset]\n", resp.Warnings)))
 	}
 
+	// Print preemptions if there are any
+	if resp.Annotations != nil && len(resp.Annotations.PreemptedAllocs) > 0 {
+		c.addPreemptions(resp)
+	}
+
 	// Print the job index info
 	c.Ui.Output(c.Colorize().Color(formatJobModifyIndex(resp.JobModifyIndex, path)))
 	return getExitCode(resp)
 }
 
+// addPreemptions shows details about preempted allocations
+func (c *JobPlanCommand) addPreemptions(resp *api.JobPlanResponse) {
+	c.Ui.Output(c.Colorize().Color("[bold][yellow]Preemptions:\n[reset]"))
+	if len(resp.Annotations.PreemptedAllocs) < preemptionDisplayThreshold {
+		var allocs []string
+		allocs = append(allocs, fmt.Sprintf("Alloc ID|Job ID|Task Group"))
+		for _, alloc := range resp.Annotations.PreemptedAllocs {
+			allocs = append(allocs, fmt.Sprintf("%s|%s|%s", alloc.ID, alloc.JobID, alloc.TaskGroup))
+		}
+		c.Ui.Output(formatList(allocs))
```

Comment:
[nit] Feel free to return early here and remove `else`. It would reduce nesting and make the big else clause easier to follow.